src/Client.ts

```diff
@@ -200,6 +217,19 @@ export default class Client {
     return client.legacyKeys.encode()
   }
 
+  private static async setupBackupClient(
+    identityKey: PrivateKey,
+    env: keyof typeof ApiUrls
+  ): Promise<BackupClient> {
+    // Hard-code the provider to use for now
+    const backupProviderSelector = async () => {
+      return Promise.resolve(
+        env === 'local' ? BackupProvider.xmtp : BackupProvider.none
```

Comment:
I'm not sure why we would only do this when connecting to `localhost`. It seems equally easy to make the backup provider a `ClientCreateOption` and just default it to none for now. 

Feels like that is going to be easier to write test cases against, and is a more explicit/simpler mental model.

Reply:
I also don't quite follow this. 

I'm not sure I understand the benefit, and it seems like this would complicate the tests by hardcoding the environment.

Reply:
I won't block it, I trust your judgement here. 

I don't particularly see a need to, keep these changes off the development network is all. (production makes total sense to me). As long as the branch is marked beta I think that is enough of a demarcation for developers. 

> For this feature specifically, the initial message backups that I make will be unencrypted, so it's really only safe to enable for localhost.

I'm less concerned about DevNetwork Beta branch data, but glad its on your mind. The XMTPLabs ran dev network can always ben wiped if needed.

🚢